pysteps/blending/steps.py

```diff
@@ -65,1723 +64,2858 @@
-                                MASK_prec_ = R_f_new > R_cmin
-                            elif mask_method == "obs":
-                                # The mask equals the most recent benchmark
-                                # rainfall field
-                                MASK_prec_ = R_pm_blended >= precip_thr
-
-                            # Set to min value outside of mask
-                            R_f_new[~MASK_prec_] = R_cmin
-
-                        # If probmatching_method is not None, resample the distribution from
-                        # both the extrapolation cascade and the model (NWP) cascade and use
-                        # that for the probability matching.
-                        if probmatching_method is not None and resample_distribution:
-                            arr1 = R_pm_ep[t_index]
-                            arr2 = precip_models_temp[j]
-                            # resample weights based on cascade level 2.
-                            # Areas where one of the fields is nan are not included.
-                            R_pm_resampled = probmatching.resample_distributions(
-                                first_array=arr1,
-                                second_array=arr2,
-                                probability_first_array=weights_pm_normalized[0],
-                            )
-                        else:
-                            R_pm_resampled = R_pm_blended.copy()
-
-                        if probmatching_method == "cdf":
-                            # nan indices in the extrapolation nowcast
-                            nan_indices = np.isnan(R_pm_ep[t_index])
-                            # Adjust the CDF of the forecast to match the resampled distribution combined from
-                            # extrapolation and model fields.
-                            # Rainfall outside the pure extrapolation domain is not taken into account.
-                            if np.any(np.isfinite(R_f_new)):
-                                R_f_new = probmatching.nonparam_match_empirical_cdf(
-                                    R_f_new, R_pm_resampled, nan_indices
-                                )
-                                R_pm_resampled = None
-                        elif probmatching_method == "mean":
-                            # Use R_pm_blended as benchmark field and
-                            mu_0 = np.mean(R_pm_resampled[R_pm_resampled >= precip_thr])
-                            MASK = R_f_new >= precip_thr
-                            mu_fct = np.mean(R_f_new[MASK])
-                            R_f_new[MASK] = R_f_new[MASK] - mu_fct + mu_0
-                            R_pm_resampled = None
+        # First determine the blending weights if method is spn. The
+        # weights for method bps have already been determined.
+        # TODO: no other weight method is possible, should we not al least give a user warning if a different weight
```

Comment:
But the BPS does nothing special then? Just to check how I would implement this in detail. Would it be something like:
if self.__config.weights_method == "spn" or self.__config.weights_method == "bps":
     if self.__config.weights_method == "spn":
            previous code
      elif self.__config.weights_method == "bps":
          pass # because nothing special happens for this one
I just added the first test since you asked for it but this is already checked in the __determine_weights_per_component() method.

Reply:
Ah sorry, I mean for the warning as mentioned in your TODO. I think good to check somewhere in the code that the right method is provided (or is that already done?) and that a warning or error is returned if another method is provided. If that is already there, you can remove this TODO.